modules/ting_search/plugins/content_types/search_result.inc

```diff
@@ -212,3 +215,32 @@ function ting_search_search_result_content_type_admin_title($subtype, $conf, $co
   $type = isset($info[$conf['type']]['title']) ? $info[$conf['type']]['title'] : t('Missing/broken type');
   return t('@type search result', array('@type' => $type));
 }
+
+
+/**
+ * Do a secondary search juggling around the drupal static.
+ *
+ * @param string $keys
+ *   The search phrase.
+ *
+ * @return Object TingClientSearchResult
+ *   The search result of the secondary search.
+ */
+function ting_search_do_secondary_search($keys) {
+
+  // Store search result temporarily. We need to make seconday search.
+  $temporary_search_result = drupal_static('ting_search_results');
+
+  // Reset the static containing search result to be able do new search.
+  drupal_static_reset('ting_search_search_execute');
+
+  // Search for the secondary keys.
+  ting_search_search_execute($keys, array('sort' => ''));
```

Comment:
If we could pass the name of the cache key we want to use here, we could get rid of a lot of this juggling around, no? Is it worth expanding the `ting_search_search_execute()` function for that?